src/directives/model/default.js

```diff
@@ -31,7 +34,7 @@ module.exports = {
     // shared setter
     function set () {
       self.set(
-        number ? _.toNumber(el.value) : el.value,
+        file ? el.files : ( number ? _.toNumber(el.value) : el.value ),
```

Comment:
It should be noted that `el.files` is a `FileList` object which is array-like, but not really an array (thus when stringifying it looks different from a stringified array). The items inside are also `File` objects. In theory you should only be putting plain objects in the vm's data, so maybe we need to process `el.files` into a proper array of plain objects here.